### PR TITLE
Fix #7120: Remove legacy logins.db and local authentication storage

### LINKED ISSUE #7120: Remove legacy logins.db
After converting the Login/Credential data be stored and used with Chromium APIs, the purpose of this cumbersome implementation of manually encrypted lost its value.
The database is only used for storing Basic Authentication Credentials which is not needed to be stored for arbitrary application launches.

In addition to clean-up the purpose of removal is to have a relief for the problem this database might be causing when application is trying to retrieve data and actually stuck while doing it, especially when opening new external urls and waking up with browser lock feature.

### PR DESCRIPTION
**Pull request by soner-yuksel**
## Summary of Changes

The PR aims to remove the local database used for login data. After converting the Login/Credential data be stored and used with Chromium APIs, the purpose of this cumbersome implementation of manually encrypted lost its value.
The database is only used for storing Basic Authentication Credentials which is not needed to be stored for arbitrary application launches.

In addition to clean-up the purpose of removal is to have a relief for the problem this database might be causing when application is trying to retrieve data and actually stuck while doing it, especially when opening new external urls and waking up with browser lock feature.

This pull request fixes #7120

Security Ticket: https://github.com/brave/security/issues/1238

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test Plan:

- Open a website that needs Basic Authentication
- Enter credentials and check everything is working with false and true credentials
-Create a new tab and open the same website and check Login Credentials is not needed

## Screenshots:

https://user-images.githubusercontent.com/6643505/228941302-3f4b6f1e-f2a1-4715-bb18-45ba3a09be28.MP4

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Storage/ThirdParty/SwiftData.swift
Comment (kylehickinson) [inline Sources/Storage/ThirdParty/SwiftData.swift:1]: With the removal of this file you should be able to fully remove `ThirdParty/sqlcipher` as well

Comment (soner-yuksel) [reply Sources/Storage/ThirdParty/SwiftData.swift:1]: I swear, I had this in my mind. :) Thanks for the warning. Fix at https://github.com/brave/brave-ios/pull/7163/commits/f0c51b278722288abd07177aed4690abf571e1e5
It is amazing to get rid of another framework

**Review by Brandon-T — APPROVED** [review]

**Review by iccub — CHANGES_REQUESTED** [review]
reminder this has to pass security review

**Comment by soner-yuksel** [comment]
> reminder this has to pass security review

Security Ticket is added to the pull request.

Security Issue: https://github.com/brave/security/issues/1238

**Review by stoletheminerals — APPROVED** [review]

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift [hunk 1]
```diff
@@ -536,6 +536,8 @@ extension BrowserViewController: WKNavigationDelegate {
         return (.cancelAuthenticationChallenge, nil)
       }
     }
+    
+    // TODO: Basic Auth Check
```
Comment (iccub) [inline Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: what about this todo?

**Review by iccub — APPROVED** [review]

File: Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift (hunk 1 above)
Comment (soner-yuksel) [reply Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift]: Ah it is actually handled right there but forgot to remove that comment later. Fix at https://github.com/brave/brave-ios/pull/7163/commits/cfbe09be4fe07a51c3b867ad92d39c66224c918a

**Pull request merged**